multistreaming-server/rtmp-conf-generator.py

```diff
@@ -73,135 +34,191 @@
+            "%%AUDIO_BIT_RATE%%",
+            transcode_config_block.get(
+                "audioBitRate", DEFAULT_AUDIO_OPTS["audioBitRate"]
+            ),
         ).replace(
-            '%%PUSH_URL%%', generatePlatormPushURL(block_config)
+            "%%AUDIO_SAMPLE_RATE%%",
+            transcode_config_block.get(
+                "audioSampleRate", DEFAULT_AUDIO_OPTS["audioSampleRate"]
+            ),
         )
-    return app_block
+    return transcode_config
 
 
-def addRTMPPushConfiguration(orig_rtmp_conf, block_config, endpoint_name):
-    if 'name' not in block_config:
-        print("ERROR - Application block is missing name element.", file=sys.stderr)
-        exit(1)
-    block_name = endpoint_name + '-' + block_config['name']
-    push_pos = orig_rtmp_conf.index('            # RTMP_PUSH_DIRECTIVE_MARKER')
-    rtmp_conf = orig_rtmp_conf[:push_pos] \
-        + '            push rtmp://localhost/' \
-        + block_name \
-        + ';\n' \
-        + orig_rtmp_conf[push_pos:]
-    block_pos = rtmp_conf.index('        # RTMP_PUSH_BLOCK_MARKER')
-    rtmp_conf = rtmp_conf[:block_pos] \
-        + createRTMPApplicationBlocks(block_name, block_config) \
-        + rtmp_conf[block_pos:]
-    return rtmp_conf
-
-
-if len(sys.argv) != 2:
-    print("Must pass a single argument of the JSON configuration file path.", file=sys.stderr)
-    sys.exit(1)
-
-try:
-    with open(sys.argv[1], 'r') as f:
-        config = json.load(f)
-except json.decoder.JSONDecodeError as err:
-    print('ERROR decoding JSON config file "{0}": {1}'.format(sys.argv[1], err), file=sys.stderr)
-    exit(1)
-except:
-    print('ERROR loading JSON config file "{0}"'.format(sys.argv[1]), file=sys.stderr)
-    exit(1)
-
-endpoint_name = config['endpoint'] if 'endpoint' in config else 'live'
-rtmp_conf = RTMP_CONF_BLOCK.replace('%%ENDPOINT_NAME%%', endpoint_name, 1)
-
-# unfortunate hack to make code compatible with previous spelling error
-config_list_key = 'rebroacastList' if 'rebroacastList' in config else 'rebroadcastList'
-for block_config in config[config_list_key]:
-    rtmp_conf = addRTMPPushConfiguration(
-            rtmp_conf, block_config, endpoint_name
+def loadJsonConfig(path):
+    try:
+        with open(path, "r") as f:
+            config = json.load(f)
+    except json.decoder.JSONDecodeError as err:
+        print(
+            'ERROR decoding JSON config file "{0}": {1}'.format(path, err),
+            file=sys.stderr,
         )
+        exit(1)
+    except:
+        print('ERROR loading JSON config file "{0}"'.format(path), file=sys.stderr)
+        exit(1)
+
+    return config
 
-print(rtmp_conf)
+
+def generateConfig(config_file, nginx_config_template):
+    config = loadJsonConfig(config_file)
+    # unfortunate hack to make code compatible with previous spelling error
+    config_list_key = (
+        "rebroacastList" if "rebroacastList" in config else "rebroadcastList"
+    )
```

Comment:
Maybe this is the version we deprecate support for my typos.

Reply:
Turns out you have the same typo on the example json config in the README! I bet a lot of people (including myself) who just pasted the example have the typo in their config right now. Might cause a big impact, but it's your call. Lmk if you still wanna go ahead with it.

Reply:
*sigh* ... ok, we will leave it.